journals/apps/journals/journal_page_helper.py

```diff
@@ -20,3 +34,74 @@ def get_nested_children(self):
 
         structure["children"] = [child.specific.get_nested_children() for child in children]
         return structure
+
+    @property
+    def site(self):
```

Comment:
Since we're having to add a `get_journal` method of `JournalAboutPage` and `JournalIndexPage` anyway, I think it'd be cleaner to just add a site property to each of these classes instead of abstracting it, since it makes `JournalIndexPage` have a method called `get_journal`.

I'd also be cool with keeping this in the mixin, but having it call a internal `self._get_site()` that each class can implement on it's own. That way we keep the high level interface, but still allow for customization.